gravitee-am-repository/gravitee-am-repository-jdbc/src/main/java/io/gravitee/am/repository/jdbc/oauth2/api/JdbcRefreshTokenRepository.java

```diff
@@ -89,21 +90,21 @@ public Completable deleteByUserId(String userId) {
     }
 
     @Override
-    public Completable deleteByDomainIdClientIdAndUserId(String domainId, String clientId, String userId) {
+    public Completable deleteByDomainIdClientIdAndUserId(String domainId, String clientId, UserId userId) {
         LOGGER.debug("deleteByDomainIdClientIdAndUserId({},{},{})", domainId, clientId, userId);
         return monoToCompletable(getTemplate().delete(JdbcRefreshToken.class)
-                .matching(Query.query(where(SUBJECT).is(userId)
+                .matching(Query.query(where(SUBJECT).is(userId.lookupSubject())
```

Comment:
**question/issue**: I'm supprised that you have to do this lookUp as the tokens (access or refresh are supposed to have the userId as reference) :thinking:

Reply:
this method is called by the revoke approvals endpoint. My understanding is that in the degraded mode, the following scenario is possible:
1. Gateway goes into degraded mode
2. User A logs in for the first time, consents to some scopes, Gateway saves the access & refresh tokens info
3. User A issues a `DELETE /consents` request (`io.gravitee.am.gateway.handler.users.UsersProvider`)
4. We need to remove token info from Gateway cache
5. Gateway goes back into "normal" mode, user A gets synchronized & receives the internal id

Reply:
So I think we either need this or need to add `userId or (source && externalId)` lookup to access & refresh tokens as well

Reply:
for now tokens (access or refresh) only rely on userId, we have to stick with it.
please don't use lookup but search only by userId

Reply:
removed the method